crates/host/src/wasmbus/mod.rs

```diff
@@ -1921,13 +1921,20 @@ impl Host {
         let host = Arc::new(host);
         let queue = spawn({
             let host = Arc::clone(&host);
-            async {
-                Abortable::new(queue, queue_abort_reg)
-                    .for_each(move |msg| {
-                        let host = Arc::clone(&host);
-                        async move { host.handle_message(msg).await }
-                    })
-                    .await;
+            async move {
+                let host = Arc::clone(&host);
+                {
+                    let host = Arc::clone(&host);
+                    Abortable::new(queue, queue_abort_reg)
+                        .for_each(move |msg| {
+                            let host = Arc::clone(&host);
+                            async move { host.handle_message(msg).await }
+                        })
+                        .await;
+                }
+                let deadline = { *host.stop_rx.borrow() };
+                host.stop_tx.send_replace(deadline);
+                info!("control interface queue task stopped");
```

Comment:
Nit: Will we only ever get here (and in the other tasks) on clean exit or could we also get here if the task aborts early? If we can abort early, should this be a `warn` or `error`?

Reply:
both and IMO it'd be pretty confusing to get a `warn`/`error` on clean exit

Reply:
We will get an error before that from the actual stream though

Reply:
To clarify: I'm wondering whether there's a case where the underlying NATS client's pollable returns a `Poll::Ready(None)` without us intentionally unsubscribing

Reply:
> I have the same question, but I think the answer is: this and the other tasks below can be stopped from receiving a signal (ctrl-C, or a `.stop` command on the control interface), or abnormally if the underlying stream finishes (which shouldn't ever happen). @rvolosatovs is there a way to handle when the stream unexpectedly closes?

did not see this before answering.
we can actually just check https://docs.rs/futures/latest/futures/stream/struct.Abortable.html#method.is_aborted
It won't always be true, there's a race condition, but it should be good enough - let me do that.

Reply:
Oo I like that idea 👍